src/Particle/LongRange/StructFact.cpp

```diff
@@ -69,19 +73,32 @@ void StructFact::resize(int ns, int nptcl, int nkpts)
 }
 
 
-void StructFact::UpdateAllPart(ParticleSet& P) { FillRhok(P); }
+void StructFact::UpdateAllPart(const ParticleSet& P)
```

Comment:
Since we have this otherwise pointless wrapper for FillRhok this would be a good place to find out what that function actually takes as input.

Reply:
The data dependency is on all the particle coordinates. So I will change the argument to DynamicCoordinates to pull ParticleSet out of this class fully. It has to be done in a separate PR.